code/controllers/subsystem/shuttles/emergency.dm

```diff
@@ -191,8 +185,25 @@
 /obj/docking_port/stationary/random/initialize()
 	..()
 	var/target_area = /area/mine/unexplored
-	var/turfs = get_area_turfs(target_area)
-	var/T=pick(turfs)
+	var/list/turfs = get_area_turfs(target_area)
+	var/sanity = 0
+	var/valid = 0
+	var/turf/T
+
+	if(!turfs.len)
+		qdel(src)
+
+	while(!valid)
+		valid = 1
+		sanity++
+		if(sanity > 100)
+			qdel(src)
+		T = pick(turfs)
+		if(!T)
+			qdel(src)
+		if(T.z == ZLEVEL_STATION || T.z == ZLEVEL_CENTCOM)
+			valid = 0
+			continue
```

Comment:
Make sure the turf its picking actually exists and is not on the station/centcomm.

Ideally this would never happen because it'd require a mapping error but Jordie asked for sanity checks